DSharpPlus/Clients/DiscordClient.WebSocket.cs

```diff
@@ -250,14 +251,14 @@ internal async Task OnInvalidateSessionAsync(bool data)
         if (data)
         {
             this.Logger.LogTrace(LoggerEvents.WebSocketReceive, "Received INVALID_SESSION (OP9, true)");
-            await Task.Delay(6000);
-            await SendResumeAsync();
+            await InternalReconnectAsync(code: 1000, message: "OP9 acknowledged");

```

Comment:
are we still resuming here

Reply:
yes, reconnect looks at the sessionID and if its present we resume

Reply:
we only not resume on reconnect if the sessionid was set to null prio to reconnecting

Reply:
To be more specific, we look at it when we get `HELLO`